### ISSUE TITLE
Integrate Namespaces

### ISSUE THREAD

**Issue by Ygg01** [Ygg01/xml5_draft#3]
I like [XML Namespaces](http://www.w3.org/TR/REC-xml-names/), they are simple and elegant, let's support them on tokenization level.

Basically extend the spec so if you encounter a `:` (not as first char of name) during parsing it starts parsing namespaces. You encounter another `:` and it will no longer consider the name fit for namespaces.

**Comment by annevk** [comment]
I think I had this implemented as well. I just looked for the first ":".

**Comment by Ygg01** [comment]
Hm, what namespaces does `<:foo>` have? Default?

**Comment by annevk** [comment]
Yeah I think you need to require at least one character before and after. Otherwise it gets hairy.

To add though, I don't think I had this in the tokenizer but rather when you create elements. You need to keep the prefix around anyway so it doesn't really make sense to do it in the tokenizer.

**Comment by Ygg01** [comment]
@annevk I was thinking way too much from my parser implementation. I think html5ever creates namespaces during tokenization.

**Comment by annevk** [comment]
But why would you do that? Also, I doubt that is the case for SVG in HTML and such.

**Comment by Ygg01** [comment]
> But why would you do that?

Well, it's implementation detail, but attributes/names in html5ever during tokenization are assigned `QualName`. Although, they are only processed in tree building phase. 

To me it made more sense initially to parse them during tokenization, since you have a `QualName` during tokenization, but now since fragments were added, I saw they are meant to be processed in tree builder phase.
